homeassistant/components/switcher_kis/sensor.py

```diff
@@ -40,92 +42,88 @@ class AttributeDescription:
     ),
 }
 
-SENSORS = {**POWER_SENSORS, **TIME_SENSORS}
+POWER_PLUG_SENSORS = POWER_SENSORS
+WATER_HEATER_SENSORS = {**POWER_SENSORS, **TIME_SENSORS}
 
 
-async def async_setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: dict,
+    config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType,
 ) -> None:
     """Set up Switcher sensor from config entry."""
-    device_data = hass.data[DOMAIN][DATA_DEVICE]
 
-    async_add_entities(
-        SwitcherSensorEntity(device_data, attribute, SENSORS[attribute])
-        for attribute in SENSORS
+    @callback
+    def async_add_sensors(wrapper: SwitcherDeviceWrapper) -> None:
+        """Add sensors from Switcher device."""
+        if wrapper.data.device_type.category == DeviceCategory.POWER_PLUG:
+            async_add_entities(
+                SwitcherSensorEntity(wrapper, attribute, POWER_PLUG_SENSORS[attribute])
+                for attribute in POWER_PLUG_SENSORS
+            )
+        elif wrapper.data.device_type.category == DeviceCategory.WATER_HEATER:
+            async_add_entities(
+                SwitcherSensorEntity(
+                    wrapper, attribute, WATER_HEATER_SENSORS[attribute]
+                )
+                for attribute in WATER_HEATER_SENSORS
+            )
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(hass, SIGNAL_DEVICE_ADD, async_add_sensors)
     )
 
 
-class SwitcherSensorEntity(SensorEntity):
+class SwitcherSensorEntity(CoordinatorEntity, SensorEntity):
     """Representation of a Switcher sensor entity."""
 
     def __init__(
         self,
-        device_data: SwitcherV2Device,
+        wrapper: SwitcherDeviceWrapper,
         attribute: str,
         description: AttributeDescription,
     ) -> None:
         """Initialize the entity."""
-        self._device_data = device_data
+        super().__init__(wrapper)
+        self.wrapper = wrapper
         self.attribute = attribute
-        self.description = description
 
         # Entity class attributes
-        self._attr_name = f"{self._device_data.name} {self.description.name}"
-        self._attr_icon = self.description.icon
-        self._attr_unit_of_measurement = self.description.unit
-        self._attr_device_class = self.description.device_class
-        self._attr_entity_registry_enabled_default = self.description.default_enabled
-        self._attr_should_poll = False
-
-        self._attr_unique_id = f"{self._device_data.device_id}-{self._device_data.mac_addr}-{self.attribute}"
+        self._attr_name = f"{wrapper.name} {description.name}"
+        self._attr_icon = description.icon
+        self._attr_unit_of_measurement = description.unit
+        self._attr_device_class = description.device_class
+        self._attr_entity_registry_enabled_default = description.default_enabled
+
+        self._attr_unique_id = f"{wrapper.device_id}-{wrapper.mac_address}-{attribute}"
```

Comment:
No need to use both device ID + mac address.

Reply:
`mac address` would be enough, but switch entities was already using both, and sensor entities which are already released also use both so changing this would be a breaking change